src/main/java/gg/projecteden/nexus/features/shops/ResourceWorldCommand.java

```diff
@@ -0,0 +1,491 @@
+import gg.projecteden.nexus.utils.MaterialTag;
+import gg.projecteden.nexus.utils.PlayerUtils;
+import gg.projecteden.nexus.utils.RandomUtils;
+import gg.projecteden.nexus.utils.Tasks;
+import gg.projecteden.nexus.utils.WorldEditUtils;
+import gg.projecteden.utils.Env;
+import gg.projecteden.utils.TimeUtils.Time;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import net.citizensnpcs.api.CitizensAPI;
+import net.citizensnpcs.api.npc.NPC;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.World.Environment;
+import org.bukkit.WorldType;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.BlockState;
+import org.bukkit.entity.Player;
+import org.bukkit.entity.TNTPrimed;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.event.block.BlockDropItemEvent;
+import org.bukkit.event.block.BlockPistonExtendEvent;
+import org.bukkit.event.block.BlockPistonRetractEvent;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.entity.EntityExplodeEvent;
+import org.bukkit.event.player.PlayerTeleportEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.math.BigDecimal;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static gg.projecteden.nexus.utils.ItemUtils.isNullOrAir;
+import static gg.projecteden.nexus.utils.WorldGroup.isResourceWorld;
+import static gg.projecteden.utils.StringUtils.prettyMoney;
+
+@NoArgsConstructor
+public class ResourceWorldCommand extends CustomCommand implements Listener {
+	private static final ResourceMarketLoggerService service = new ResourceMarketLoggerService();
+	private ResourceMarketLogger logger;
+
+	public ResourceWorldCommand(@NonNull CommandEvent event) {
+		super(event);
+		if (isPlayerCommandEvent())
+			logger = getLogger(world());
+	}
+
+	static {
+		World survival = Bukkit.getWorld("survival");
+		World resource = Bukkit.getWorld("resource");
+		if (survival != null && resource != null)
+			resource.setMonsterSpawnLimit((int) (survival.getMonsterSpawnLimit() * 1.5));
+	}
+
+	@Path
+	void warp() {
+		runCommand("warp resource");
+	}
+
+	@Confirm
+	@Permission("group.admin")
+	@Path("reset <test>")
+	void reset(boolean test) {
+		resetWorlds(test);
+	}
+
+	@Confirm
+	@Permission("group.admin")
+	@Path("setup <test>")
+	void setup(boolean test) {
+		setupWorlds(test);
+	}
+
+	private ResourceMarketLogger getLogger(World world) {
+		if (!isResourceWorld(world))
+			throw new InvalidInputException("Not allowed outside of resource world");
```

Comment:
This is used in event handlers, while theoretically it could work, I try to avoid using anything from the commands framework, because most of the time it wont

Reply:
gotcha 👍🏻